src/BoundaryConditions/field_boundary_conditions.jl

```diff
@@ -14,47 +14,50 @@ Construct a `FieldBoundaryConditions` using a `CoordinateBoundaryCondition` for
 """
 FieldBoundaryConditions(x, y, z) = FieldBoundaryConditions((x, y, z))
 
-"""
-    HorizontallyPeriodicBCs(;   top = BoundaryCondition(Flux, nothing),
-                             bottom = BoundaryCondition(Flux, nothing))
-
-Construct `FieldBoundaryConditions` with `Periodic` boundary conditions in the x and y
-directions and specified `top` (+z) and `bottom` (-z) boundary conditions for u, v,
-and tracer fields.
-
-`HorizontallyPeriodicBCs` cannot be applied to the the vertical velocity w.
-"""
-function HorizontallyPeriodicBCs(;    top = BoundaryCondition(Flux, nothing),
-                                   bottom = BoundaryCondition(Flux, nothing))
-
-    x = PeriodicBCs()
-    y = PeriodicBCs()
-    z = CoordinateBoundaryConditions(bottom, top)
-
-    return FieldBoundaryConditions(x, y, z)
-end
-
-"""
-    ChannelBCs(; north = BoundaryCondition(Flux, nothing),
-                 south = BoundaryCondition(Flux, nothing),
-                   top = BoundaryCondition(Flux, nothing),
-                bottom = BoundaryCondition(Flux, nothing))
-
-Construct `FieldBoundaryConditions` with `Periodic` boundary conditions in the x
-direction and specified `north` (+y), `south` (-y), `top` (+z) and `bottom` (-z)
-boundary conditions for u, v, and tracer fields.
-
-`ChannelBCs` cannot be applied to the the vertical velocity w.
-"""
-function ChannelBCs(;  north = BoundaryCondition(Flux, nothing),
-                       south = BoundaryCondition(Flux, nothing),
-                         top = BoundaryCondition(Flux, nothing),
-                      bottom = BoundaryCondition(Flux, nothing)
-                    )
-
-    x = PeriodicBCs()
-    y = CoordinateBoundaryConditions(south, north)
-    z = CoordinateBoundaryConditions(bottom, top)
+default_bc(::Periodic) = PeriodicBC()
+default_bc(::Bounded)  = NoFluxBC()
+
+valid_directions = (:west, :east, :south, :north, :bottom, :top)
+
+function FieldBoundaryConditions(grid; kwargs...)
+    kws = keys(kwargs)
+    for kw in kws
+        if kw ∉ valid_directions
+            e = "$kw is not a valid keyword argument. Must be one of " *
+                "the valid directions: $valid_directions"
+            throw(ArgumentError(e))
+        end
+    end
+
+    TX, TY, TZ = topology(grid)
+
+    if (TX isa Periodic || TX isa Flat) && (:west in kws || :east in kws)
+        e = "Cannot specify west or east boundary conditions with $TX topology in x-direction."
+        throw(ArgumentError(e))
+    else
+        west_bc = :west in kws ? kwargs[:west] : default_bc(TX)
```

Comment:
could let default_bc take two arguments: The topology and the current bc. Then define a fallback that throws an error, whitelist the bcs that are valid, and define a default for ::Nothing. Might be a bit simpler and more modular, less repeated code.